File: 2441-largest-positive-integer-that-exists-with-its-negative/2441-largest-positive-integer-that-exists-with-its-negative.cpp
class Solution {
public:
    int findMaxK(vector<int>& nums) {
        unordered_map<int, int>ump;
        for(int i=0; i<nums.size(); i++)
        {
            ump[nums[i]]++;
        }
        
        int maxValue = INT_MIN;
        bool ok = 0;
        for(int i=0; i<nums.size(); i++)
        {
            auto it = ump.find(-nums[i]);
            if(it != ump.end())
            {
                ok = 1;
                if(nums[i]>maxValue)
                {
                    maxValue = nums[i];
                }
            }  
        }
        
        return(ok == 1 ? maxValue : -1);
    }
};

File: 2441-largest-positive-integer-that-exists-with-its-negative/2441-largest-positive-integer-that-exists-with-its-negative.cs
public class Solution {
    public int FindMaxK(int[] nums) {
        Dictionary<int, int> ump = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++) {
            if (ump.ContainsKey(nums[i])) {
                ump[nums[i]]++;
            } else {
                ump[nums[i]] = 1;
            }
        }

        int maxValue = int.MinValue;
        bool ok = false;
        for (int i = 0; i < nums.Length; i++) {
            if (ump.ContainsKey(-nums[i])) {
                ok = true;
                if (nums[i] > maxValue) {
                    maxValue = nums[i];
                }
            }
        }

        return (ok ? maxValue : -1);
    }
}